ygen/enumgen.go

```diff
@@ -61,29 +60,66 @@ type enumGenState struct {
 	// a module such as openconfig-bgp which defines /bgp and is also used at
 	// /network-instances/network-instance/protocols/protocol/bgp.
 	uniqueEnumeratedLeafNames map[string]string
+
+	// The flags below are cached findEnumSet parameters that are required for
+	// re-calculating the keys used to query the name sets.
+	compressPaths bool
```

Comment:
Yes, there is a reason other than argument length: the unique enum key used to generate *and* retrieve the enums depend on these flags; so, it never makes sense for these flags to deviate from what was first used (e.g. requesting a non-compressed version of an enum-leaf when everything was generated with compress=true). In essence, each `enumSet` is specific to these flags.

This makes the generation less flexible as it prevents the flags from differing according to each entry, but considering that we never practically change these around within the same generation instance, I thought the reduced flexibility introduced wasn't a real problem. Even if the flexibility is needed in the future, some work will be needed to allow this in the new "generate-all-at-once" world, because the current mechanism doesn't know which ones need the different flags; so, I think keeping the current code simple makes sense.

Reply:
Sure - but that's true throughout the code, the configuration that is supplied to start with must be immutable, otherwise the code generation will not make any sense. We didn't choose to hand around the configuration type throughout because of this (i.e., we didn't say that we must use the receiver of the configuration for every element). It seems inconsistent to allow other methods that depend on having consistency in these flags to take this as an argument, and then in this case have it stored in the receiver.

I'd prefer that we keep this explicitly as an argument to the function so that it's clear that this is a function that is dependent upon the values that are specified for these flags (and which of them we are actually dependent on). With that approach it also becomes explicitly clear when we add a new value upon which the function output is conditional - whereas in this approach if I add a new field to `enumState`, it's not clear which of the downstream functions depend on it without further examination of the implementation.

Reply:
Got it, makes sense.